handlers/whatsapp/whatsapp_test.go

```diff
@@ -122,21 +263,99 @@ var defaultSendTestCases = []ChannelSendTestCase{
 	{Label: "Error",
 		Text: "Error", URN: "whatsapp:250788123123",
 		Status:       "E",
-		ResponseBody: `{ "errors": [{"title":"Error Sendind"}] }`, ResponseStatus: 403,
+		ResponseBody: `{ "errors": [{ "title": "Error Sending" }] }`, ResponseStatus: 403,
 		RequestBody: `{"to":"250788123123","type":"text","text":{"body":"Error"}}`,
+		Error:       "Received error from send endpoint: Error Sending",
 		SendPrep:    setSendURL},
 	{Label: "No Message ID",
 		Text: "Error", URN: "whatsapp:250788123123",
 		Status:       "E",
 		ResponseBody: `{ "messages": [] }`, ResponseStatus: 200,
 		RequestBody: `{"to":"250788123123","type":"text","text":{"body":"Error"}}`,
+		Error:       "Unable to get message id from response body",
 		SendPrep:    setSendURL},
 	{Label: "Error Field",
 		Text: "Error", URN: "whatsapp:250788123123",
 		Status:       "E",
-		ResponseBody: `{ "errors": [{"title":"Error Sendind"}] }`, ResponseStatus: 200,
+		ResponseBody: `{ "errors": [{"title":"Error Sending"}] }`, ResponseStatus: 200,
 		RequestBody: `{"to":"250788123123","type":"text","text":{"body":"Error"}}`,
+		Error:       "Received error from send endpoint: Error Sending",
 		SendPrep:    setSendURL},
+	{Label: "Audio Send",
+		Text:   "audio has no caption",
+		URN:    "whatsapp:250788123123",
+		Status: "W", ExternalID: "157b5e14568e8",
+		Attachments: []string{"audio/mpeg:https://foo.bar/audio.mp3"},
+		ResponseBodies: map[MockedRequest]MockedResponse{
```

Comment:
@norkans7 I'm not hugely happy about this attribute name, `ResponseBodies`, perhaps `Responses` is better? Any other input from you on this change. It's a fairly substantial change to your test harness and, if this is a route you're happy to take, probably can take some further refactoring to clean up the `ExternalID`, `ResponseStatus`, `ResponseBody` and `RequestBody` test case attributes.

Reply:
And yes that will require some refactor and clean up for those attributes